### ISSUE TITLE
hyphens missing from some prefixes, suffixes, etc.

### ISSUE THREAD

**Issue by robaki** [tatuylonen/wiktextract#463]
When processing preextracted data (json downloaded from here https://kaikki.org/dictionary/Finnish/index.html) I noticed that in etymology sections sometimes suffixes, prefixes, etc. have the hyphens and sometimes not. Long story short, wiktionary adds the missing hyphens when rendering a page, but these hyphens are missing in around 900 entries in the pages' code (that's finnish only),  and as a consequence, the extracted datasets.

Here is what I was told when I asked adding the missing hyphens it on wiki:

> This sounds like a parsing/logic issue on your end - {{[suf](https://en.wiktionary.org/wiki/Template:suf)}} and {{[suffix](https://en.wiktionary.org/wiki/Template:suffix)}} automatically add a leading hyphen, if missing, to all entries after the first one. (Likewise, {{[pre](https://en.wiktionary.org/wiki/Template:pre)}} and {{[prefix](https://en.wiktionary.org/wiki/Template:prefix)}} automatically add a trailing hyphen, if missing, to all entries except the last one.

Not sure if you're interested in wiktextract handling this - if yes, I can look into it and contribute.

**Comment by robaki** [comment]
sure, here is one example with missing hyphen:

https://en.wiktionary.org/wiki/kyseenalaistaa

rendered page:
> Etymology
> [kyseenalainen](https://en.wiktionary.org/wiki/kyseenalainen#Finnish) +‎ [-taa](https://en.wiktionary.org/wiki/-taa#Finnish)

code (`taa` should have a hyphen):
```
===Etymology===
{{suffix|fi|kyseenalainen|taa}} 
```

and one that doesn't have this problem:

https://en.wiktionary.org/wiki/hiillostaa
> Etymology
> [hiillos](https://en.wiktionary.org/wiki/hiillos#Finnish) +‎ [-taa](https://en.wiktionary.org/wiki/-taa#Finnish)

```
===Etymology===
{{affix|fi|hiillos|-taa}}
```

**Comment by robaki** [comment]
some more words with "taa" instead of "-taa":
> 'säätää', 'panostaa', 'salvostaa', 'vedostaa', 'naurunalaistaa', 'veistää', 'hammastaa', 'niveltää', 'piirtää'

and there are other affixes with this issue: I have found the following while doing some manual screening:
> 'ia', 'la', 'in', 'ko', 'uu', 'nti', 'ke', 'ti', 'ele', 'su'

**Comment by xxyzz** [comment]
Sorry I don't understand the problem, the expanded template text `etymology_text` is correct: "kyseenalainen + -taa` on kaikki.org: https://kaikki.org/dictionary/All%20languages%20combined/meaning/k/ky/kyseenalaistaa.html

**Comment by robaki** [comment]
OK, I think I understand it now - I expected the `args` field to be consistent, but this is "raw" input from wiktionary, so it just inherits their problems.

Thanks, I'll need to change my approach then and either use the `expansion` field or add the hyphens myself.

**Comment by kristian-clausal** [comment]
Yeah, the args field is just the parameters/arguments in the template, so if it's `|taa|`, it's just `taa`. The template takes care of it by ignoring the hyphen, I guess. If the output is consistent (which it should be in this case), you can parse or regex the `-taa` from `+ -taa` pretty consistently, though!
